tests/test_openclhpp.cpp

```diff
@@ -1665,6 +1665,33 @@ void testKernelSetArgLocal()
     kernelPool[0].setArg(2, cl::Local(123));
 }
 
+void testKernelSetArgBySetKernelArgSVMPointerWithUniquePtrType()
+{
+#if CL_HPP_TARGET_OPENCL_VERSION >= 200
+    std::unique_ptr<int> buffer(new int(1000));
+    clSetKernelArgSVMPointer_ExpectAndReturn(make_kernel(0), 1, buffer.get(), CL_SUCCESS);
+    TEST_ASSERT_EQUAL(kernelPool[0].setArg(1, buffer), CL_SUCCESS);
+#endif
+}
+
+void testKernelSetArgBySetKernelArgSVMPointerWithVectorType()
+{
+#if CL_HPP_TARGET_OPENCL_VERSION >= 200
+    VECTOR_CLASS<int> vec(1000);
+    clSetKernelArgSVMPointer_ExpectAndReturn(make_kernel(1), 2, vec.data(), CL_SUCCESS);
+    TEST_ASSERT_EQUAL(kernelPool[1].setArg(2, vec), CL_SUCCESS);
+#endif
+}
+
+void testKernelSetArgBySetKernelArgSVMPointerWithPointerType()
+{
+#if CL_HPP_TARGET_OPENCL_VERSION >= 200
+    cl_mem *memory = &bufferPool[1]();
```

Comment:
This could also pass any random pointer here too, including e.g. the address of a stack variable, but this should be fine too.